core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -485,4 +485,13 @@ package object config {
         "array in the sorter.")
       .intConf
       .createWithDefault(Integer.MAX_VALUE)
+
+  private[spark] val SHUFFLE_MAP_OUTPUT_PARALLEL_AGGREGATION_THRESHOLD =
+    ConfigBuilder("spark.shuffle.mapOutput.parallelAggregationThreshold")
+      .internal()
+      .doc("Multi-thread is used when the number of mappers * shuffle partitions is greater than " +
+        "or equal to this threshold.")
```

Comment:
Is this condition to enable parallel aggregation still true?

Reply:
Sorry, but didn't get you.

Reply:
Looks like only `parallelism` >= 2, this parallel aggregation is enabled. Is it equal to `the number of mappers * shuffle partitions >= this threshold`?

Reply:
From above `statuses.length.toLong * totalSizes.length / parallelAggThreshold + 1`, looks like we need to have at least two times of this threshold to enable this parallel aggregation?

Reply:
`statuses.length.toLong * totalSizes.length / parallelAggThreshold + 1` >= 2 -> `statuses.length.toLong * totalSizes.length >= parallelAggThreshold`, so it doesn't need to be 2 times, just not smaller than 1x is good.

Reply:
Says `statuses.length.toLong * totalSizes.length` is `10010000`, for example:

```scala
scala> 10010000 / 10000001
res0: Int = 1
```

Now, it is more than the threshold, but the parallel aggregation is not enabled...

Reply:
Oh, I see...nvm. I misread the `+ 1`...

Reply:
Do you think it's necessary to indicate the actual parallelism's calculation way here?

Reply:
It's ok. I misread the equation. Nvm.

Reply:
I think we don't need to indicate the calculation way in config description. The current one is enough.

Reply:
After rethinking about this, I think it is better to indicate this threshold also determines the number of threads in parallelism. So it should not be set to zero or negative number.

Reply:
Yeah, I will add some.